javascript/src/components/streams/StreamControls.jsx

```diff
@@ -0,0 +1,86 @@
+/* global jsRoutes */
+
+'use strict';
+
+var React = require('react/addons');
+var DropdownButton = require('react-bootstrap').DropdownButton;
+var MenuItem = require('react-bootstrap').MenuItem;
+var ButtonGroup = require('react-bootstrap').ButtonGroup;
+var StreamForm = require('./StreamForm');
+var PermissionsMixin = require('../../util/PermissionsMixin');
+var UsersStore = require('../../stores/users/UsersStore');
+
+var StreamControls = React.createClass({
+    mixins: [PermissionsMixin],
+    getInitialState() {
+        return {};
+    },
+    _onResume(evt) {
+        this.props.onResume(this.props.stream);
+        this.refs.dropdownButton.setDropdownState(false);
+    },
+    _onPause(evt) {
+        this.props.onPause(this.props.stream);
+        this.refs.dropdownButton.setDropdownState(false);
+    },
+    _onEdit(evt) {
+        this.refs.streamForm.open();
+    },
+    _onClone(evt) {
+        this.refs.cloneForm.open();
+    },
+    _onCloneSubmit(streamId, stream) {
+        this.props.onClone(this.props.stream.id, stream);
+    },
+    _onQuickAdd() {
+        this.props.onQuickAdd(this.props.stream.id);
+    },
+    componentDidMount() {
+        UsersStore.load(this.props.username).done((user) => {
+            this.setState({user: user});
+        });
+    },
+    render() {
+        var permissions = this.props.permissions;
+        var stream = this.props.stream;
+
+        var menuItems = [];
+
+        if (this.isPermitted(permissions, ['streams:edit:' + stream.id])) {
+            menuItems.push(<MenuItem key={"editStreams-" + stream.id}><a onClick={this._onEdit}>Edit stream</a></MenuItem>);
```

Comment:
Adding an anchor as a children of `MenuItem` results in empty options in the dropdown menu (you can see them while moving slowly over it). Please add the `onClick` event to the MenuItem and only pass text as children of `MenuItem`.